include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
- * after the message was successfully passed to the TLS layer.  In the case of QoS 1
- * the function returns after the receipt of the PUBACK control packet.
+ * This function sends an MQTT message to the server. The server will then
+ * forward this message to any clients with subscriptions to topic filters
+ * that match the message's topic.
  *
- * @param pClient Reference to the IoT Client
- * @param pTopicName Topic Name to publish to
+ * For a QoS 0 message, this function returns after the message is successfully
+ * passed to the TLS layer. For a QoS 1 message, this function returns after the
+ * receipt of the PUBACK for the transmitted message.
+ *
+ * @param pClient MQTT client context
+ * @param pTopicName Topic name to publish to
  * @param topicNameLen Length of the topic name
- * @param pParams Pointer to Publish Message parameters
+ * @param pParams Publish message parameters
  *
- * @return An IoT Error Type defining successful/failed publish
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_publish] */
 IoT_Error_t aws_iot_mqtt_publish(AWS_IoT_Client *pClient, const char *pTopicName, uint16_t topicNameLen,
 								 IoT_Publish_Message_Params *pParams);
+/* @[declare_mqtt_publish] */
 
 /**
  * @brief Subscribe to an MQTT topic.
  *
- * Called to send a subscribe message to the broker requesting a subscription
- * to an MQTT topic.
- * @note Call is blocking.  The call returns after the receipt of the SUBACK control packet.
- * @warning pTopicName and pApplicationHandlerData need to be static in memory.
+ * This function sends an MQTT subscribe packet to the server. It registers
+ * a subscription that will cause the provided callback function to be invoked
+ * when the server sends a matching message to the client.
  *
- * @param pClient Reference to the IoT Client
- * @param pTopicName Topic Name to publish to. pTopicName needs to be static in memory since
- * no malloc is performed by the SDK
- * @param topicNameLen Length of the topic name
- * @param qos Quality of service for subscription
- * @param pApplicationHandler Reference to the handler function for this subscription
- * @param pApplicationHandlerData Point to data passed to the callback. pApplicationHandlerData
- * also needs to be static in memory since no malloc is performed by the SDK
+ * @note Incoming messages are handled by @ref mqtt_function_yield. Therefore,
+ * @ref mqtt_function_yield must always be called regularly if any subscriptions
+ * are active.
  *
- * @return An IoT Error Type defining successful/failed subscription
+ * @param[in] pClient MQTT client context
+ * @param[in] pTopicName Topic for subscription
+ * @param[in] topicNameLen Length of topic
+ * @param[in] qos Quality of service for subscription
+ * @param[in] pApplicationHandler Callback function for incoming messages that arrive
+ * on this subscription.
+ * @param[in] pApplicationHandlerData Data passed to the callback.
```

Comment:
(Nit) _Context_ passed to the callback?

Reply:
The variable name ends in "Data", so I called it "Data" for consistency.